File: src/main/java/ejercicioJava/sistemaReservaHotel/GestorReservas.java
package ejercicioJava.sistemaReservaHotel;

import java.util.ArrayList;
import java.util.List;

public class GestorReservas {

    public List<Reserva> reservas = new ArrayList<>();

    public boolean reservarHabitacion(Cliente cliente, Habitacion habitacion) {
        if (cliente.getHabitacion() == null && habitacion.isDisponibilidad()) {
            cliente.setHabitacion(habitacion);
            habitacion.setDisponibilidad(false);
            Reserva reserva = new Reserva(cliente, habitacion);
            cliente.setReserva(reserva);
            reservas.add(reserva);
            return true;
        }
        return false;
    }

    public void cancelarReserva(Cliente cliente) {
        if (cliente.getHabitacion() != null) {
            cliente.getHabitacion().setDisponibilidad(true);
        }
        reservas.remove(cliente.getReserva());
        cliente.setReserva(null);
    }

    public void mostrarReservas() {
        for (Reserva reserva : reservas) {
            System.out.println(reserva);
        }
    }
}

File: src/main/kotlin/ejercicioKotlin/sistemaReservaHotel/GestorReservas.kt
package org.example.sistemaReservaHotel

class GestorReservas {

    private val reservas:MutableList<Reserva> = mutableListOf()

    fun reservarHabitacion(cliente: Cliente, habitacion: Habitacion):Boolean {
        if(cliente.habitacion == null && habitacion.disponibilidad) {
            cliente.habitacion = habitacion
            habitacion.disponibilidad = false
            val reserva = Reserva(cliente, habitacion)
            cliente.reserva = reserva
            reservas.add(reserva)
            return true
        }
        return false
    }

    fun cancelarRerva(cliente: Cliente) {
        if (cliente.habitacion != null) cliente.habitacion!!.disponibilidad = true
        reservas.remove(cliente.reserva)
        cliente.reserva = null
    }

    fun mostrarReservas() {
        reservas.forEach { println(it) }
    }

}